test/parallel/test-stream-readable-default-encoding.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { Readable } = require('stream');
+
+{
+  assert.throws(() => {
+    new Readable({
+      read: () => {},
+      defaultEncoding: 'my invalid encoding',
+    });
+  }, {
+    code: 'ERR_UNKNOWN_ENCODING',
+  });
+}
+
+{
+  const r = new Readable({
+    read() {},
+    defaultEncoding: 'hex'
+  });
+
+  r.push('ab');
+
+  const chunks = [];
+  r.on('data', (chunk) => chunks.push(chunk));
```

Comment:
Nit: you are pushing a single chunk so there is no need to use an array of chunks and `process.nextTick()`. You can just add the assertion here.